thunder/__init__.py

```diff
@@ -576,9 +586,18 @@ def apply_transforms_and_build_cache_entry(cd, cs, cache_info, prologue_trc, com
                     executors_list=cd.executors_list,
                     use_del_last_used=False,
                 )
-                computation_traces.extend(extraces)
-                computation_trc = computation_traces[-1]
-                computation_trc = thunder.executors.passes.del_last_used(computation_trc)
+                computation_trc = extraces[-1]
```

Comment:
Note that the joint traces are not being appended here, and that is also delayed until after the splitting.  I'm unsure whether this is desired behavior or not.

Reply:
No, we would prefer to append things early if only to look at them when transforms error, but we fix this separately, it's not that great in the current torch_autograd code either.